src/averageAges.js

```diff
@@ -37,7 +50,22 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let women = people.filter(person => person.sex === 'f');
+
+  if (withChildren) {
+    const someoneMothers = people.map(person => person.mother);
+
+    const mothersInArr = people.filter((person) =>
+      someoneMothers.includes(person.name));
+
+    women = mothersInArr;
+  }
+
+  const ageSum = women.reduce((sum, current) => {
+    return sum + (current.died - current.born);
+  }, 0);
+
+  return ageSum / women.length;
 }
```

Comment:
We have same lines in all three functions, think of creating a separate function for it